pkgs/tools/package-management/libdnf/default.nix

```diff
@@ -0,0 +1,59 @@
+{ gcc9Stdenv, stdenv, fetchFromGitHub, cmake, gettext, pkgconfig, gpgme, libsolv, openssl, check, pcre, json_c, libmodulemd, libsmartcols, sqlite, librepo, libyaml, rpm }:
+
+gcc9Stdenv.mkDerivation rec {
+  pname = "libdnf";
+  version = "0.55.2";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = pname;
+    rev = version;
+    sha256 = "0hiydwfa90nsrqk5ffa6ks1g73wnsgjgq7z7gwq9jj76a7gpfbfq";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gettext
+    pkgconfig
+    check
+  ];
+
+  propagatedBuildInputs = [
+    gpgme
+    openssl
+    json_c
+    sqlite
+    librepo
+    pcre
+    rpm
+    libsolv
+    libsmartcols
+    libyaml
+    libmodulemd
+  ];
+
+  patches = stdenv.lib.optionals stdenv.isDarwin [ ./darwin.patch ];
+
+  prePatch = ''
+    # According to #cmake on freenode, libdnf should bundle the FindLibSolv.cmake module
+    cp ${libsolv}/share/cmake/Modules/FindLibSolv.cmake cmake/modules/
+  '';
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace "enable_testing()" "" \
+      --replace "add_subdirectory(tests)" ""
+  '';
+
+  configurePhase = ''
+    cmake . -DCMAKE_INSTALL_PREFIX=$out -DWITH_ZCHUNK=OFF -DWITH_GTKDOC=OFF -DWITH_HTML=OFF -DWITH_BINDINGS=OFF
```

Comment:
Please use ``cmakeFlags`` instead of overwriting the configurePhase.

Reply:
Hm, If I change it to:
```
 cmakeFlags = [
   "-DCMAKE_INSTALL_PREFIX=$out"
   "-DWITH_GTKDOC=OFF"
   "-DWITH_HTML=OFF"
   "-DWITH_BINDINGS=OFF"
 ] ++ stdenv.lib.optional stdenv.isDarwin [ "-DWITH_ZCHUNK=OFF" ]; 
```

I end up with compile errors like this one:

```
make[2]: *** [dnf/CMakeFiles/microdnf.dir/build.make:314: dnf/CMakeFiles/microdnf.dir/plugins/repoquery/dnf-command-repoquery.c.o] Error 1
/tmp/nix-build-microdnf-3.6.0.drv-9/source/dnf/dnf-main.c:27:10: fatal error: 'libdnf/libdnf.h' file not found
```

Trying to figure out why

Reply:
Try dropping `"-DCMAKE_INSTALL_PREFIX=$out"`, `$out` is not valid for CMake interpolation, and the setup hook already sets that argument: https://github.com/NixOS/nixpkgs/blob/1e6e7c2bef96107dcab3e172a65bf44bee018f1b/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L34

Reply:
Though I would try using Meson as that is what the spec file lists.

Reply:
Ha, it was DCMAKE_INSTALL_PREFIX. Now it works :)